File: PassBYValueReference/modifyValues.cpp
#include <iostream>

// Function to modify a variable passed by value
void modifyByValue(int num) {
    num = 42;
}

// Function to modify a variable passed by reference
void modifyByReference(int& num) {
    num = 42;
}

int main() {
    int number = 7;

    // Pass 'number' by value
    modifyByValue(number);
    std::cout << "Value after modifyByValue: " << number << std::endl; // Output: 7

    // Pass 'number' by reference
    modifyByReference(number);
    std::cout << "Value after modifyByReference: " << number << std::endl; // Output: 42

    return 0;
}

File: PassBYValueReference/modifyValues.py
class MyClass:
    def __init__(self, value):
        self.value = value

# Function to modify a variable passed by value
def modify_by_value(num):
    num = 42

# Function to modify an object passed by reference
def modify_by_reference(obj):
    obj.value = 42

number = 7

# Pass 'number' by value
modify_by_value(number)
print("Value after modify_by_value:", number)  # Output: 7

my_obj = MyClass(7)

# Pass 'my_obj' by reference
modify_by_reference(my_obj)
print("Value after modify_by_reference:", my_obj.value)  # Output: 42
